payment/datastore.go

```diff
@@ -0,0 +1,354 @@
+	statement := "SELECT * FROM transactions WHERE order_id = $1"
+	transactions := []Transaction{}
+	err := pg.DB.Select(&transactions, statement, orderID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &transactions, nil
+}
+
+// GetTransaction returns a single of transaction given an external transactin Id
+func (pg *Postgres) GetTransaction(externalTransactionID string) (*Transaction, error) {
+	statement := "SELECT * FROM transactions WHERE external_transaction_id = $1"
+	transaction := Transaction{}
+	err := pg.DB.Get(&transaction, statement, externalTransactionID)
+
+	if err == sql.ErrNoRows {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	return &transaction, nil
+}
+
+// UpdateOrder updates the orders status.
+// 	Status should either be one of pending, paid, fulfilled, or canceled.
+func (pg *Postgres) UpdateOrder(orderID uuid.UUID, status string) error {
+	_, err := pg.DB.Exec(`UPDATE orders set status = $1, updated_at = CURRENT_TIMESTAMP where id = $2`, status, orderID)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateTransaction creates a transaction given an orderID, externalTransactionID, currency, and a kind of transaction
+func (pg *Postgres) CreateTransaction(orderID uuid.UUID, externalTransactionID string, status string, currency string, kind string, amount decimal.Decimal) (*Transaction, error) {
+	tx := pg.DB.MustBegin()
+
+	var transaction Transaction
+	err := tx.Get(&transaction,
+		`
+			INSERT INTO transactions (order_id, external_transaction_id, status, currency, kind, amount)
+			VALUES ($1, $2, $3, $4, $5, $6)
+			RETURNING *
+	`, orderID, externalTransactionID, status, currency, kind, amount)
+
+	if err != nil {
+		_ = tx.Rollback()
+		return nil, err
+	}
+
+	err = tx.Commit()
+
+	if err != nil {
+		_ = tx.Rollback()
+		return nil, err
+	}
+
+	return &transaction, nil
+}
+
+// GetSumForTransactions returns the calculated sum
+func (pg *Postgres) GetSumForTransactions(orderID uuid.UUID) (decimal.Decimal, error) {
+	var sum decimal.Decimal
+
+	err := pg.DB.Get(&sum, `
+		SELECT SUM(amount) as sum
+		FROM transactions
+		WHERE order_id = $1 AND status = 'completed'
+	`, orderID)
+
+	return sum, err
+}
+
+// InsertIssuer inserts the given issuer
+func (pg *Postgres) InsertIssuer(issuer *Issuer) (*Issuer, error) {
+	statement := `
+	insert into order_cred_issuers (merchant_id, public_key)
+	values ($1, $2)
+	returning *`
+	issuers := []Issuer{}
+	err := pg.DB.Select(&issuers, statement, issuer.MerchantID, issuer.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(issuers) != 1 {
+		return nil, errors.New("Unexpected number of issuers returned")
+	}
+
+	return &issuers[0], nil
+}
+
+// GetIssuer retrieves the given issuer
+func (pg *Postgres) GetIssuer(merchantID string) (*Issuer, error) {
+	statement := "select * from order_cred_issuers where merchant_id = $1"
+	issuers := []Issuer{}
+	err := pg.DB.Select(&issuers, statement, merchantID)
```

Comment:
Could you update places within this file where we probably should be using `Get` to retrieve a single item?

Reply:
https://github.com/brave-intl/bat-go/pull/226/commits/7e6a1355064f3196d65ff4f32a85fcafec2fa78b